lib/template-mixins.js

```diff
@@ -113,7 +113,12 @@ module.exports = function (fields, options) {
                 hintId: extension.hintId || (hint ? key + '-hint' : null),
                 error: this.errors && this.errors[key],
                 maxlength: maxlength(key) || extension.maxlength,
-                required: fields[key] && fields[key].required !== undefined ? fields[key].required : true,
+                required: !fields[key]
+                  ? false
+                  : (fields[key].required !== undefined)
+                    ? fields[key].required
+                    : (fields[key].validate && (_.indexOf(fields[key].validate, 'required') !== -1))
+                      ? true : false,
                 pattern: extension.pattern,
```

Comment:
Would a "var" do? Or do you want to get rid of one or two of the IF expressions in favour of IF statements?